### ISSUE TITLE
Build fails after adding Charts

### ISSUE THREAD

**Issue by wcrama** [project-serum/serum-dex-ui#95]
Hi all,

After following the steps in readme and have been adding the folders charting_library datafeeds to /src and /public i receive errors on building.
What did i do wrong here ?

Failed to compile.

/Applications/XAMPP/xamppfiles/htdocs/serum-dex-ui/src/components/TradingView/index.tsx
TypeScript error in /Applications/XAMPP/xamppfiles/htdocs/serum-dex-ui/src/components/TradingView/index.tsx(52,11):
Property 'container' is missing in type '{ symbol: string; datafeed: any; interval: Nominal<string, "ResolutionString">; container_id: string | undefined; library_path: string; locale: "en"; disabled_features: string[]; ... 7 more ...; theme: "Dark"; }' but required in type 'ChartingLibraryWidgetOptions'.  TS2741

    50 | 
    51 |   React.useEffect(() => {
  > 52 |     const widgetOptions: ChartingLibraryWidgetOptions = {
       |           ^
    53 |       symbol: findTVMarketFromAddress(
    54 |         market?.address.toBase58() || '',
    55 |       ) as string,

**Comment by pepeneif** [comment]
I have the same issue.
I think it might have to do with the version of the Tradingview charting-library, but not 100% sure yet.
If I find the answer, I will post it here.
If you find the answer, please share it here.

**Comment by wcrama** [comment]
Issue is in TradingView Chart charting_library.d.ts:

interface ChartingLibraryWidgetOption

remove container

but... the chart is still not displaying.. we're missing some key element here.

**Comment by pepeneif** [comment]
I will try with an older version of the Trading View charting_library, maybe that is the solution.

My delay is because the access to the Trading View charting_library is using my business partner github user and not mine, maybe you can try yourself?

Choose a charting_library version few months old to be sure.

To get an older version of any software in github click on "Comits" on the top right corner of the files list.

**Comment by wcrama** [comment]
Let me try...

**Comment by wcrama** [comment]
replaced with Release v18.044 (from c3634edc) …
No diff, for some reason there is no return for <div id={defaultProps.containerId} className="tradingview-chart" />;

**Comment by wcrama** [comment]
div id={defaultProps.containerId} className="tradingview-chart"

**Comment by wcrama** [comment]
1. Download serum-dex-ui
2. Download Trading Charts
3. Copy `charting_library` folder from https://github.com/tradingview/charting_library/ to `/public` and to `/src` folders.
4. Copy `datafeeds` folder from https://github.com/tradingview/charting_library/ to `/public`.
5. Add import { TVChartContainer } from '../components/TradingView'; too TradePage.tsx
6. Removed "./src/components/TradingView/index.tsx" from tsconfig.json
7. Uncomment in public/index.html so script tags are enabled
8. Created a file .eslintignore -> src/charting_library/charting_library.esm.js and placed in root (together with .env.example)
9. Removed container variable from ChartingLibraryWidgetOptions
10. 

But no chart is being returned…. What am i missing here ?

**Comment by pepeneif** [comment]
You need to add some code to pages/TradePage.tsx

Inside the "const RenderNormal" definition

Something like this:

const RenderNormal = ({ onChangeOrderRef, onPrice, onSize }) => {
  return (
    <Row style={{minHeight: '900px', flexWrap: 'nowrap', }}>

      <Col flex={'360px'} style={{ height: '100%' }}>
        <Orderbook smallScreen={false} onPrice={onPrice} onSize={onSize} />
        <TradesTable smallScreen={false} />
      </Col>

      <Col flex="auto" style={{ height: '100%', display: 'flex', flexDirection: 'column' }}>
        <div style={{ height: '510px' }}><TVChartContainer /></div>
        <UserInfoTable />
      </Col>

      <Col flex="400px" style={{ height: '100%', display: 'flex', flexDirection: 'column' }}>
        <TradeForm setChangeOrderRef={onChangeOrderRef} />
        <StandaloneBalancesDisplay />
      </Col>

    </Row>
  );
};

But depending on your design, the order of the windows and sizes can vary.

**Comment by wcrama** [comment]
That is already inside that file:
```
const RenderNormal = ({ onChangeOrderRef, onPrice, onSize }) => {
  return (
    <Row
      style={{
        minHeight: '900px',
        flexWrap: 'nowrap',
      }}
    >
      <Col flex="auto" style={{ height: '100%', display: 'flex' }}>
        <UserInfoTable />
      </Col>
      <Col flex={'360px'} style={{ height: '100%' }}>
        <Orderbook smallScreen={false} onPrice={onPrice} onSize={onSize} />
        <TradesTable smallScreen={false} />
      </Col>
      <Col
        flex="400px"
        style={{ height: '100%', display: 'flex', flexDirection: 'column' }}
      >
        <TradeForm setChangeOrderRef={onChangeOrderRef} />
        <StandaloneBalancesDisplay />
      </Col>
    </Row>
  );
};
```

**Comment by pepeneif** [comment]
But if you see, in the code that is already in the file, the TVChartContainer is missing.

**Comment by wcrama** [comment]
> But if you see, in the code that is already in the file, the TVChartContainer is missing.

? Can you explain this please.

**Comment by wcrama** [comment]
you have to add 
import { TVChartContainer } from '../components/TradingView';

manually in TradePage.tsx

**Comment by pepeneif** [comment]
yes, you need to import  TVChartContainer , as it is indicated in the instructions.

But then,

you also have to include that container somewhere in the rendering of the page.

In the code I sent to you it will show between the TradesTable and the TradeForm, on top of UserInfoTable

In that page you can change the arranging of the windows shown.

**Comment by wcrama** [comment]
yeah thats why the compiler now returnss:

Compiled with warnings.

./src/pages/TradePage.tsx
  Line 28:10:  'TVChartContainer' is defined but never used  @typescript-eslint/no-unused-vars
  
  
  let me check you last comment and get back to you.

**Comment by wcrama** [comment]
Compiled with warnings.

./src/components/TradingView/index.tsx
  Line 96:6:  React Hook React.useEffect has missing dependencies: 'defaultProps.autosize', 'defaultProps.clientId', 'defaultProps.containerId', 'defaultProps.datafeedUrl', 'defaultProps.fullscreen', 'defaultProps.interval', 'defaultProps.libraryPath', 'defaultProps.studiesOverrides', and 'defaultProps.userId'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

**Comment by pepeneif** [comment]
and do you see the chart now?

**Comment by wcrama** [comment]
Yes it is showing now ! Not the right dimensions or chart data but we're one step further !
i'll check if it works with the latests trading view chart aswell.

**Comment by pepeneif** [comment]
Great!

The dimensions are changed in the <div style={{ height: '510px' ... next to the <TVChartContainer /> tag as per the JS/React flex display instructions.

I have the same problem about the chart data and I was thinking it was the charting_library version.

If you find the solution, please share it?

**Comment by wcrama** [comment]
sure! thnx pepeneif

**Comment by wcrama** [comment]
it is a CORS issue..

**Comment by pepeneif** [comment]
Wow.
Thanks for letting me know!!

**Comment by LeRoYGo** [comment]
I did what you wrote here and got this
![image](https://user-images.githubusercontent.com/56492696/123849881-c31d4680-d932-11eb-84e9-a24b86d7246a.png)

**Comment by pepeneif** [comment]
Maybe you forgot step number 8 of the list given by wcrama ?

8. Created a file .eslintignore ->  and placed in root (together with .env.example)

it means that you have to create a file called ".eslintignore", with "src/charting_library/charting_library.esm.js" as its content.

**Comment by LeRoYGo** [comment]
![image](https://user-images.githubusercontent.com/56492696/123855256-20b49180-d939-11eb-8c6f-4f1d78e7f2c4.png)
Like this?

**Comment by pepeneif** [comment]
yes

**Comment by wcrama** [comment]
> Wow.
> Thanks for letting me know!!

change -> export const BONFIDA_DATA_FEED inside bonfidaConnector
This will make the chart work ! bonfida api does not work atm

**Comment by wcrama** [comment]
> ![image](https://user-images.githubusercontent.com/56492696/123855256-20b49180-d939-11eb-8c6f-4f1d78e7f2c4.png)
> Like this?

add these 2 lines
src/charting_library/charting_library.esm.js
src/components/TradingView/index.tsx

**Comment by wcrama** [comment]
otherwise try older versions of tradinview from commits in github

**Comment by pepeneif** [comment]
Which version of the charting_library worked for you wcrama ?

**Comment by wcrama** [comment]
Commits on May 26, 2021
Release v18.044 (from c3634edc) …

**Comment by pepeneif** [comment]
Great, thanks!

**Comment by wcrama** [comment]
i also have this line in charting_library.esm.js
/* eslint-disable no-unused-expressions */
![Screenshot 2021-06-30 at 17 51 19](https://user-images.githubusercontent.com/33249073/123992410-e21be700-d9cb-11eb-8f89-6384f878ae6f.png)

**Comment by LeRoYGo** [comment]
I did as written here, but the graphs are not displayed. The terminal says the following:
`Compiled with warnings.

./src/charting_library/charting_library.esm.js
  Line 102:70:  Unexpected use of comma operator  no-sequences
  Line 137:65:  Unexpected use of comma operator  no-sequences
  Line 204:59:  Unexpected use of comma operator  no-sequences
  Line 317:24:  Unexpected use of comma operator  no-sequences
  Line 321:63:  Unexpected use of comma operator  no-sequences
  Line 326:49:  Unexpected use of comma operator  no-sequences
  Line 375:72:  Unexpected use of comma operator  no-sequences
  Line 479:48:  Unexpected use of comma operator  no-sequences

./src/components/TradingView/index.tsx
  Line 96:6:  React Hook React.useEffect has missing dependencies: 'defaultProps.autosize', 'defaultProps.clientId', 'defaultProps.containerId', 'defaultProps.datafeedUrl', 'defaultProps.fullscreen', 'defaultProps.interval', 'defaultProps.libraryPath', 'defaultProps.studiesOverrides', and 'defaultProps.userId'. Either include them or remove the dependency array  react-hooks/exhaustive-deps`

**Comment by LeRoYGo** [comment]
![image](https://user-images.githubusercontent.com/56492696/124298670-f6034c80-db75-11eb-9be5-f79d3a91b87c.png)

Hurrah, the diagram is working, but there are two problems:
1) The selected trading pair is different from the pair in the chart. If you change the pair in DEX on the diagram does not change.
2) How to make diagrams in height more. The method above does not work.

**Comment by wcrama** [comment]
Good work ! Would you like to share how you managed ?

**Comment by richovercash** [comment]
I followed the instructions above but got a failed to compile error. Anyone know how to fix this?
./src/components/TradingView/index.tsx
Identifier 'TVChartContainer' has already been declared. (36:13)

  34 | export interface ChartContainerState {}
  35 |
 36 | export const TVChartContainer = () => {

  37 |   // let datafeed = useTvDataFeed();
  38 |   const defaultProps: ChartContainerProps = {
  39 |     symbol: 'BTC/USDC',

**Comment by dahifi** [comment]
I was able to setup new instance using most recent serum and TV library, just need to add the `/* eslint-disable no-unused-expressions */` in front of ./src/charting_library/charting_library.esm.js.

**Comment by DefiDax** [comment]
can anyone send me the charting_library please or a link to find it? thanks

**Comment by DefiDax** [comment]
> I was able to setup new instance using most recent serum and TV library, just need to add the `/* eslint-disable no-unused-expressions */` in front of ./src/charting_library/charting_library.esm.js.

Hi could you send me a copy of the charting library please??

**Comment by DefiDax** [comment]
> Good work ! Would you like to share how you managed ?

can anyone send me the charting_library please or a link to find it? Been driving me mad last 2 weeks and cant find it nowhere thanks.. email cryptoc1980@gmail.com

**Comment by interbiznw** [comment]
> > Good work ! Would you like to share how you managed ?
> 
> can anyone send me the charting_library please or a link to find it? Been driving me mad last 2 weeks and cant find it nowhere thanks.. email [cryptoc1980@gmail.com](mailto:cryptoc1980@gmail.com)

Read the instructions, you need to request access from tradingview directly

**Comment by BunnyNomics101** [comment]
> Good work ! Would you like to share how you managed ?

scrams would care to share with us how you got amstadex working?
We could compensate: Bunnynomics101@gmail.com

**Comment by interbiznw** [comment]
> > Good work ! Would you like to share how you managed ?
> 
> scrams would care to share with us how you got amstadex working? We could compensate: [Bunnynomics101@gmail.com](mailto:Bunnynomics101@gmail.com)

You need help?

**Comment by wcrama** [comment]
Not sure if anyone still need help but I'm happy to assist !

**Comment by BunnyNomics101** [comment]
> Not sure if anyone still need help but I'm happy to assist ! 

Yeah man. We need help, please email us

**Comment by BunnyNomics101** [comment]
> > > Good work ! Would you like to share how you managed ?
> > 
> > 
> > scrams would care to share with us how you got amstadex working? We could compensate: [Bunnynomics101@gmail.com](mailto:Bunnynomics101@gmail.com)
> 
> You need help?

Yeah, we are so close with our code! It took us months to resolve the issues: https://github.com/BunnyNomics101/Sobull.git

**Comment by BunnyNomics101** [comment]
Did you ever get pass the "Error placing order Transaction failed: Instruction: InitializeAccount" when placing orders?

**Comment by Prazovic** [comment]
> change -> export const BONFIDA_DATA_FEED inside bonfidaConnector

Hi Bro, change to what? The original version: export const BONFIDA_DATA_FEED = 'https://serum-api.bonfida.com/tv'; 
Thank you for your help!

**Comment by rapitex** [comment]
Hi every one
In my case solved by adding this comment to top ===/* eslint-disable */ ==== of esm.js file in charting library and also changing the file construct by using baeutify contetnt extension in IDE.
